tests/test_time_series_mapper.py

```diff
@@ -0,0 +1,149 @@
+
+
+def generate_data(one_week=False) -> pd.DataFrame:
+    def generate__one_week_per_month_by_hour():
+        dfd = pd.DataFrame(
+            {
+                "id": np.repeat(1, 12 * 24 * 7),
+                "month": np.repeat(range(1, 13), 24 * 7),
+                "day_of_week": np.tile(
+                    np.repeat(range(7), 24), 12
+                ),  # 0: Monday, 6: Sunday, ~ pyspark.weekday(), duckdb.isodow()-1, pd.day_of_week
+                "hour": np.tile(range(24), 12 * 7),
+            }
+        )
+        dfd["value"] = dfd["month"] * 1000 + dfd["day_of_week"] * 100 + dfd["hour"]
+        dfd = pd.concat([dfd, dfd.assign(id=2).assign(value=dfd.value * 2)], axis=0)
+        return dfd
+
+    def generate__one_weekday_day_and_one_weekend_day_per_month_by_hour():
+        dfd = pd.DataFrame(
+            {
+                "id": np.repeat(1, 12 * 24 * 2),
+                "month": np.repeat(range(1, 13), 24 * 2),
+                "is_weekday": np.tile(np.repeat([True, False], 24), 12),
+                "hour": np.tile(range(24), 12 * 2),
+            }
+        )
+        dfd["value"] = dfd["month"] * 1000 + dfd["is_weekday"] * 100 + dfd["hour"]
+        dfd = pd.concat([dfd, dfd.assign(id=2).assign(value=dfd.value * 2)], axis=0)
+        return dfd
+
+    if one_week:
+        dfd = generate__one_week_per_month_by_hour()
+    else:
+        dfd = generate__one_weekday_day_and_one_weekend_day_per_month_by_hour()
+
+    # Add more data
+    df = pd.DataFrame(
+        {
+            "id": [1, 2],
+            "geography": ["IL", "CO"],
+            "time_zone": ["US/Central", "US/Mountain"],
+        }
+    )
+    dfd = dfd.merge(df, on="id", how="left")
+
+    return dfd
+
+
+def get_destination_schema(tzinfo) -> TableSchema:
+    schema = TableSchema(
+        name="mapped_data",
+        time_config=DatetimeRange(
+            start=datetime(year=2018, month=1, day=1, tzinfo=tzinfo),
+            resolution=timedelta(hours=1),
+            length=8760,
+            interval_type=TimeIntervalType.PERIOD_BEGINNING,
+            time_column="timestamp",
+        ),
+        time_array_id_columns=["id", "geography", "time_zone"],
+        value_column="value",
+    )
+    return schema
+
+
+def get_data_schema(one_week: bool = False) -> TableSchema:
+    if one_week:
+        time_format = RepresentativePeriodFormat.ONE_WEEK_PER_MONTH_BY_HOUR
+    else:
+        time_format = (
+            RepresentativePeriodFormat.ONE_WEEKDAY_DAY_AND_ONE_WEEKEND_DAY_PER_MONTH_BY_HOUR
+        )
+    schema = TableSchema(
+        name="load_data",
+        time_config=RepresentativePeriodTimeRange(
+            interval_type=TimeIntervalType.PERIOD_BEGINNING,
+            time_format=time_format,
+        ),
+        time_array_id_columns=["id", "geography", "time_zone"],
+        value_column="value",
+    )
+    return schema
+
+
+def get_timeseries(time_config: DatetimeRange) -> pd.Series:
```

Comment:
Should this function be a test of `DatetimeRange.list_timestamps`? And then from that time on we call our method?